astropy/modeling/mappings.py

```diff
@@ -10,16 +10,34 @@
 
 
 class Mapping(Model):
-    def __init__(self, mapping):
+    """
+    Allows inputs to be reordered, duplicated or dropped.
+
+    Parameters
+    ----------
+    mapping : tuple
+        Integers representing indices of the inputs.
+
+    Raises
+    ------
+    TypeError
+        Raised when number of inputs is less that `max(mapping)`.
+
+    Examples
+    --------
+    >>> poly1 = Polynomial2D(1, c0_0=1, c1_0=2, c0_1=3)
+    >>> poly2 = Polynomial2D(1, c0_0=1, c1_0=2.4, c0_1=2.1)
+    >>> model = (Shift(1) & Shift(2)) | Mapping((0, 1, 0, 1)) | (poly1 & poly2)
+    >>> model(1, 2)
+    (17.0, 14.2)
+
+    """
+    def __init__(self, mapping, **kwargs):
```

Comment:
What `kwargs` does this actually take?  There are several arguments accepted by models in general that don't apply to this.  If we want to support things like `name` and `meta` that's fine, but I'd rather list them explicitly (since other things like `model_set_axis` don't apply).